singularity-eos/eos/eos_builder.cpp

```diff
@@ -28,11 +28,15 @@ EOS EOSBuilder::buildEOS(EOSBuilder::EOSType type,
   bool scaled  = (modifiers.count(EOSModifier::Scaled) > 0);
   bool shifted = (modifiers.count(EOSModifier::Shifted) > 0);
   bool relativistic = (modifiers.count(EOSModifier::Relativistic) > 0);
-  if ((shifted || scaled || relativistic) && !(isModifiable(type))) {
+  bool units = (modifiers.count(EOSModifier::UnitSystem) > 0);
+  if ((shifted || scaled || relativistic || units) && !(isModifiable(type))) {
     EOS_ERROR("Modifiers not supported for this EOS");
   }
-  if (relativistic && (shifted || scaled)) {
-    EOS_ERROR("Relativistic modifier cannot be combined with shift or scale");
+  if (relativistic && (shifted || scaled || units)) {
+    EOS_ERROR("Relativistic modifier currently must be applied alone");
+  }
+  if (units && (shifted || scaled || relativistic)) {
+    EOS_ERROR("Unit System modifier currently must be applied alone");
```

Comment:
What would need to change to combine the unit system modifier with other modifiers? Units are only modifying input and output values, right, so I don't see how it would interact with other modifiers (I'm not a singularity modifiers expert though)